chevah/github-hooks-server
# [#7] [functional-test] Have the hook server control the review labels

Scope
======

This PR is here to implement label management for the review process.

The idea is that we will continue to use our normal review workflows and the GitHub hooks server will auto-set the labels so that we can filter the reviews

Changes
======

When a review is requested, all reviewers are set as assignees for the PR.
In this way, we know which reviewers are required, and which are optional... and which of the required reviewers have not yet reviewed the PR. The needs-review label is set and all other labels are removed.

When changes are requested, the needs-review  (or needs-merge) label is removed and needs-changes is set. The PR requester is set as assignee.

When changes are approved


This adds git-secret to manage the private testing data.

--------

actions can be done as comments and as dedicated review actions.

needs-review can only be requested as a comment.



This is used for testing the review process.

This is used by the automated tests for the functional tests. Don't close it and make sure you unsubscribe from it.

When a review is requested, the reviewers are set as assignees.
When a review is done, they are removed.

The labels are also updated.

In this way, it should be easier to generate a report.


How to test
=========

reviewers: @hcs0 @lgheorghiu @dumol 

manually set various labels and see how they are updated after an action.

Trigger various actions and see how the labels/assignees are auto updated.
